sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -3,209 +3,253 @@
+    :keyword selects: List of setting selectors to filter configuration settings
+    :type selects: list[~azure.appconfigurationprovider.SettingSelector]
+    :keyword trim_prefixes: List of prefixes to trim from configuration keys
+    :type trim_prefixes: list[str]
+    :keyword key_vault_options: Options for resolving Key Vault references
+    :type key_vault_options: ~azure.appconfigurationprovider.KeyVaultOptions
+    """
+    #pylint:disable=protected-access
+    key_vault_options = kwargs.pop("key_vault_options", None)
 
-        provider._trim_prefixes = sorted(kwargs.pop("trimmed_key_prefixes", []), key=len, reverse=True)
+    connection_string = kwargs.pop("connection_string", None)
+    endpoint = kwargs.pop("endpoint", None)
+    credential = kwargs.pop("credential", None)
 
-        provider._dict = {}
+    provider = __buildprovider(connection_string, endpoint, credential, key_vault_options)
 
-        secret_clients = key_vault_options.secret_clients if key_vault_options else {}
+    selects = kwargs.pop("selects", {SettingSelector("*", "\0")})
 
-        for select in selects:
-            configurations = provider._client.list_configuration_settings(
-                key_filter=select.key_filter, label_filter=select.label_filter
-            )
-            for config in configurations:
+    provider._trim_prefixes = sorted(kwargs.pop("trimmed_key_prefixes", []), key=len, reverse=True)
 
-                trimmed_key = config.key
-                # Trim the key if it starts with one of the prefixes provided
-                for trim in provider._trim_prefixes:
-                    if config.key.startswith(trim):
-                        trimmed_key = config.key[len(trim) :]
-                        break
+    secret_clients = key_vault_options.secret_clients if key_vault_options else {}
 
-                if config.content_type == KEY_VAULT_REFERENCE_CONTENT_TYPE:
-                    secret = provider.__resolve_keyvault_reference(config, key_vault_options, secret_clients)
-                    provider._dict[trimmed_key] = secret
-                elif provider.__is_json_content_type(config.content_type):
-                    try:
-                        j_object = json.loads(config.value)
-                        provider._dict[trimmed_key] = j_object
-                    except json.JSONDecodeError:
-                        # If the value is not a valid JSON, treat it like regular string value
-                        provider._dict[trimmed_key] = config.value
-                else:
+    for select in selects:
+        configurations = provider._client.list_configuration_settings(
+            key_filter=select.key_filter, label_filter=select.label_filter
+        )
+        for config in configurations:
+
+            trimmed_key = config.key
+            # Trim the key if it starts with one of the prefixes provided
+            for trim in provider._trim_prefixes:
```

Comment:
In .NET, we process KeyVault/Json types first, then trim prefixes. Can we do the same here to maintain the same behavior? Changing the order might produce different results, especially when we start supporting feature flags.

Reply:
If you look down a bit, the trimmed_key is only used in the dictionary, `_resolve_keyvault_reference` uses the `config` which contains the untrimmed key.

Reply:
Just thinking about how feature flags would look in future - we'd want to maintain the same schema that we use in .net and spring providers, right? So for a feature "Beta", dictionary key would be something like "FeatureManagement:Beta" (or whatever separator makes sense for python).  
Since feature flag flattening changes the key being added to dictionary, we shouldn't use trimmed_key as dictionary key.

Reply:
I see two options:

```python
elif config.content_type == FeatureFlagConfigurationSetting._feature_flag_content_type: #pylint: disable=protected-access
    feature_management = provider._dict.get("FeatureManagement", {})
    feature_management[trimmed_key] = config.value
```

or

```python
elif config.content_type == FeatureFlagConfigurationSetting._feature_flag_content_type: #pylint: disable=protected-access
    provider._dict["FeatureManagement." + trimmed_key] = config.value
```

Reply:
Discussed with Matt offline. Since the config dictionary structure is not limited to <string,string> in python, what you have right now should be good.